### ISSUE TITLE
Allow repository administrators to create or edit submission forms directly in the user interface

### ISSUE THREAD

**Issue by bjpot** [DSpace/DSpace#8804]
We are utilizing a DSpace 5 instance and are exploring the possibility of migrating to DSpace 7.

Over the past year, our federated repository structure has undergone significant changes. The administrator function has been decentralized to different community levels, resulting in some difficulties in implementing change controls at the central library.

In DSpace 7, it would be beneficial to have a feature that enables an overall repository administrator to easily modify submission forms from the user interface. This feature should be user-friendly and accessible, such as a form-like interface, a UI data-frame, or an exported spreadsheet. This should be achievable without needing to access an XML file in the Configuration directory on the server or contracting a third-party to make the changes.

A colleague has informed me that other DSpace repositories in our region, such as https://repository.escholarship.umassmed.edu/, have customized features built into their XMLUI interface. Their administrators can update or create submission forms via an uploaded spreadsheet. We believe it would be useful if this custom feature could be integrated as a core function within DSpace Angular.

**Comment by tdonohue** [comment]
Agreed that this would be a great feature.  If someone already has a way to achieve this, we'd love for the code or concepts to be shared back... DSpace is very much a collaboration.  Most of the code / new features are contributed via community members or service providers. 

In the meantime, this needs a volunteer to investigate or contribute code back.  I'll leave this ticket open for others to comment on or provide suggestions.

**Comment by saschaszott** [comment]
I am also interested in this ticket. Ideally, the submission form configuration should be manageable directly through the admin UI in the browser. As a first step, I would like to discuss whether we should store the submission form configuration natively in the database and allow r+w access via REST API. In that case, the XML file would simply remain as an additional configuration option. We would need an XML importer, and the import could potentially happen on the fly. The admin UI could read the submission form configuration from the database (via REST API) and write any changes back. However, one would then need to ensure later on that the XML file is updated with the current submission form configuration before making further modifications to it. In the long term, the XML file could be phased out entirely.
